arrow-fx-coroutines/src/main/kotlin/arrow/fx/coroutines/stream/callbacks.kt

```diff
@@ -0,0 +1,143 @@
+package arrow.fx.coroutines.stream
+
+import arrow.core.Either
+import arrow.fx.coroutines.CancelToken
+import arrow.fx.coroutines.ExitCase
+import arrow.fx.coroutines.ForkConnected
+import arrow.fx.coroutines.Promise
+import arrow.fx.coroutines.UnsafePromise
+import arrow.fx.coroutines.andThen
+import arrow.fx.coroutines.guaranteeCase
+import arrow.fx.coroutines.stream.concurrent.Queue
+import kotlin.coroutines.Continuation
+import kotlin.coroutines.EmptyCoroutineContext
+import kotlin.coroutines.startCoroutine
+import kotlin.experimental.ExperimentalTypeInference
+
+private object END
+
+interface EmitterSyntax<A> {
+  fun emit(a: A): Unit
+  fun emit(chunk: Chunk<A>): Unit
+  fun emit(iterable: Iterable<A>): Unit
```

Comment:
Should we include an option for emitting a `Stream<A>`?

Reply:
Could potentially emit with two strategies:
- emit: waits for the end of the stream before continuing
- emitPar: pushes the stream up in parallel

The end result would be similar to concatenate and merge operations respectively

Reply:
I believe one could achieve the same by just combining that possible Stream with the callback itself, no?

* `Stream.callback { ... }.merge(Stream.xxx)` 
* `Stream.callback { ... }.flatMap(Stream.xxx).parJoinUnbounded()`

Reply:
That's ok if we have knowledge of the other stream outside. My thinking was if we require to do conditions inside the callback... 
So, depending on the value we get from a given callback we could start one or another.

One use case I can think of for the secuencial emit could be, if we have a lifecycle callback that generates a new stream of user events each time, ending the previous one. This way we could generate a seamless continuous stream of user events.

As for the parallel emit we could have something that listens to multiple UI elements (like selectable items) and would mean we can combine them all into a single stream.

Not a very good example that last one as it could be done as a merge...

Let's create a separate ticket and I'll investigate more useful use cases or drop it if we find it not that worth it :)

Reply:
Moved here: https://github.com/arrow-kt/arrow-fx/issues/240